storage/storage.go

```diff
@@ -427,4 +427,5 @@ type DeviceToken struct {
 	Expiry              time.Time
 	LastRequestTime     time.Time
 	PollIntervalSeconds int
+	PKCE                PKCE
```

Comment:
It seems that changes to the database schema are required.  Otherwise, PKCE settings will not persist. 
It would also be nice to add [a conformance test](https://github.com/dexidp/dex/blob/f49e7bc218ebffdc5bfb800c53c06d273958eef2/storage/conformance/conformance.go#L991-L1032) for the storage.

Pieces of a code that should be changed (and the corresponding Get and Update functions too):
https://github.com/dexidp/dex/blob/f49e7bc218ebffdc5bfb800c53c06d273958eef2/storage/ent/client/devicetoken.go#L11-L25

https://github.com/dexidp/dex/blob/f49e7bc218ebffdc5bfb800c53c06d273958eef2/storage/kubernetes/types.go#L814-L831

https://github.com/dexidp/dex/blob/f49e7bc218ebffdc5bfb800c53c06d273958eef2/storage/sql/crud.go#L927-L944

https://github.com/dexidp/dex/blob/f49e7bc218ebffdc5bfb800c53c06d273958eef2/storage/etcd/types.go#L286-L295